esp_bsp_generic/Kconfig

```diff
@@ -0,0 +1,467 @@
+           
+            config BSP_BUTTON_5_ADC_VALUE
+                depends on BSP_BUTTON_5_TYPE_ADC
+                int "Button 5 ADC Value"
+                default 2000
+                range 0 2500
+                help
+                    The ADC middle value for button 5.  
+        endmenu
+    endmenu
+
+    menu "uSD card - Virtual File System"
+        menu "Pins"
+            depends on SOC_SDMMC_USE_GPIO_MATRIX
+            config BSP_SD_CLK
+                int "uSD GPIO CLK"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD CLK.
+            config BSP_SD_CMD
+                int "uSD GPIO CMD"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD CMD.
+            config BSP_SD_D0
+                int "uSD GPIO D0"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD D0.
+            config BSP_SD_D1
+                int "uSD GPIO D1"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD D1.
+            config BSP_SD_D2
+                int "uSD GPIO D2"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD D2.
+            config BSP_SD_D3
+                int "uSD GPIO D3"
+                default -1
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The GPIO pin for uSD D3.
+            
+        endmenu
+        config BSP_SD_FORMAT_ON_MOUNT_FAIL
+            bool "Format uSD card if mounting fails"
+            default n
+            help
+                The SDMMC host will format (FAT) the uSD card if it fails to mount the filesystem.
+
+        config BSP_SD_MOUNT_POINT
+            string "uSD card mount point"
+            default "/sdcard"
+            help
+                Mount point of the uSD card in the Virtual File System
+
+        config BSP_SD_MAX_FILES
+            int "Max files supported for SD VFS"
+            default 2
+            help
+                Supported max files for SD in the Virtual File System.
+
+    endmenu
+
+    menu "SPIFFS - Virtual File System"
+        config BSP_SPIFFS_FORMAT_ON_MOUNT_FAIL
+            bool "Format SPIFFS if mounting fails"
+            default n
+            help
+                Format SPIFFS if it fails to mount the filesystem.
+
+        config BSP_SPIFFS_MOUNT_POINT
+            string "SPIFFS mount point"
+            default "/spiffs"
+            help
+                Mount point of SPIFFS in the Virtual File System.
+
+        config BSP_SPIFFS_PARTITION_LABEL
+            string "Partition label of SPIFFS"
+            default "storage"
```

Comment:
This will fail to compile withou a partition table a `PARTITION_TABLE_CUSTOM ` option set. Can we make this part of this BSP?

Reply:
We have it same in all BSPs. I think, that example (or application) should set it, when want to use SPIFFS.